DataFormats/L1TMuon/interface/RegionalMuonCand.h

```diff
@@ -201,7 +201,7 @@ namespace l1t {
 
   private:
     int m_hwPt;
-    int m_hwPt2;
+    int m_hwPtUnconstrained;
```

Comment:
@silviodonato You're referring to this change, right? If the new `m_hwPtUnconstrained` corresponds to the old `m_hwPt2` (i.e. it is essentially a rename), then indeed an iorule would be good to keep the old files fully readable. If they have different intended behavior (such that old `m_hwPt2` can not be translated to `m_hwPtUnconstrained` in any meaningful way), I suspect our only option is to allow a crack in the backwards compatibility.

On the other hand, if it would be the case that `m_hwPt2` would be known the `0` by construction before (for some reasion), I'm not sure if an iorule would really be worth it.

Reply:
Yes, I'm referring to this change. My understanding is that `m_hwPt2` is simply renamed to `m_hwPtUnconstrained`.

> On the other hand, if it would be the case that m_hwPt2 would be known the 0 by construction before (for some reasion), I'm not sure if an iorule would really be worth it.

I agree. @rekovic @dinyar do you expect `m_hwPt2` is zero all existing data (RelVals, etc...)?

Reply:
Hi @makortel, @silviodonato,

ah, thanks for the explanation! It is indeed a rename in order to make the field a bit more descriptive. I haven't used iorules before, would something like the following work?

```
<ioread sourceClass="l1t::RegionalMuonCand" targetClass="l1t::RegionalMuonCand" version="[]" target="m_hwPtUnconstrained" source="int m_hwPt2" embed="false">
  <![CDATA[ m_hwPtUnconstrained = onfile.m_hwPt2; ]]>
</ioread>
```

However while searching for the applicable versions it seems that RegionalMuonCand doesn't have an entry in `DataFormats/L1TMuon/src/classes_def.xml`.. 

I'm not sure I understand your last sentence, @makortel, but in principle m_hwPt2 was filled in previous CMSSW releases as far as I'm aware. Maybe it's best to revert the rename of that field? The main objective was in any case to make the getters and setters more descriptive which if I understand correctly can be maintained.

Reply:
> However while searching for the applicable versions it seems that RegionalMuonCand doesn't have an entry in `DataFormats/L1TMuon/src/classes_def.xml`..

Itself it indeed isn't, but a derived class `L1MuBMTrack` has an entry
https://github.com/cms-sw/cmssw/blob/e3a2e63fcfb011c2b053e921dd088e0cc3d9e1b1/DataFormats/L1TMuon/src/classes_def.xml#L68-L73
(which is also modified in this PR following the change).

I'm actually not sure what to do in case a base class is modified but does not have an explicit entry in `classes_def.xml`. @Dr15Jones @pcanal could you help?

Reply:
Unless the base class is "empty" (no persistent) data, you should generate a dictionary for it.  If it is empty (and there is no complaint as-is) then I guess nothing need to be done.

Reply:
Thanks Philippe. That means (IIUC) that an entry for the original `l1t::RegionalMuonCand` should be added in the `classes_def.xml`, probably starting with the class version 10. Then the entry should be updated with this update, and an iorule along (slightly modified from https://github.com/cms-sw/cmssw/pull/31608#discussion_r501528032)
```xml
<ioread sourceClass="l1t::RegionalMuonCand" targetClass="l1t::RegionalMuonCand" version="[-10]"  source="int m_hwPt2" target="m_hwPtUnconstrained">
  <![CDATA[ m_hwPtUnconstrained = onfile.m_hwPt2; ]]>
</ioread>
```

Reply:
We are seeing segmentation faults in the IB when trying to read back this class.